nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/apache/nifi/processors/kite/InferAvroSchema.java

```diff
@@ -0,0 +1,454 @@
+            }
+        });
+
+        return avroSchema;
+    }
+
+    /**
+     * Infers the Avro schema from the input Flowfile content.
+     *
+     * @param inputFlowFile
+     *  The original input FlowFile containing the JSON content as it entered this processor.
+     *
+     * @param context
+     *  ProcessContext to pull processor configurations.
+     *
+     * @param session
+     *  ProcessSession to transfer FlowFiles
+     */
+    private AtomicReference<String> inferAvroSchemaFromJSON(final FlowFile inputFlowFile, final ProcessContext context, final ProcessSession session) {
+
+        final AtomicReference<String> avroSchema = new AtomicReference<>();
+        session.read(inputFlowFile, new InputStreamCallback() {
+            @Override
+            public void process(InputStream in) throws IOException {
+                Schema as = JsonUtil.inferSchema(
+                        in, context.getProperty(RECORD_NAME).evaluateAttributeExpressions().getValue(),
+                        context.getProperty(NUM_RECORDS_TO_ANALYZE).evaluateAttributeExpressions().asInteger());
+                avroSchema.set(as.toString(context.getProperty(PRETTY_AVRO_OUTPUT).asBoolean()));
+
+            }
+        });
+
+        return avroSchema;
+    }
+
+    /**
+     * Examines the incoming FlowFiles mime.type attribute to determine if the schema should be inferred for CSV or JSON data.
+     *
+     * @param inputFlowFile
+     *  The original input FlowFile containing the content.
+     *
+     * @param context
+     *  ProcessContext to pull processor configurations.
+     *
+     * @param session
+     *  ProcessSession to transfer FlowFiles
+     */
+    private AtomicReference<String> inferAvroSchemaFromMimeType(final FlowFile inputFlowFile, final ProcessContext context, final ProcessSession session) {
+
+        String mimeType = inputFlowFile.getAttribute(CoreAttributes.MIME_TYPE.key());
+        AtomicReference<String> avroSchema = new AtomicReference<>();
+
+        if (mimeType!= null) {
+            switch (mimeType) {
+                case JSON_MIME_TYPE:
+                    getLogger().debug("Inferred content type as JSON from \"{}\" value of \"{}\"", new Object[]{CoreAttributes.MIME_TYPE.key(),
+                            inputFlowFile.getAttribute(CoreAttributes.MIME_TYPE.key())});
+                    avroSchema = inferAvroSchemaFromJSON(inputFlowFile, context, session);
+                    break;
+                case CSV_MIME_TYPE:
+                    getLogger().debug("Inferred content type as CSV from \"{}\" value of \"{}\"", new Object[]{CoreAttributes.MIME_TYPE.key(),
+                            inputFlowFile.getAttribute(CoreAttributes.MIME_TYPE.key())});
+                    avroSchema = inferAvroSchemaFromCSV(inputFlowFile, context, session);
+                    break;
+                default:
+                    getLogger().warn("{} ", new Object[]{CoreAttributes.MIME_TYPE.key()});
```

Comment:
This log message is missing any context - perhaps change to:
getLogger().warn("Unable to infer Avro Schema from {} because its mime type is {}, which is not supported by this Processor", new Object[] {inputFlowFile, mimeType} );